pkg/preflights/host-preflight.yaml

```diff
@@ -116,6 +116,13 @@ spec:
     - udpPortStatus:
         collectorName: Calico Communication Port
         port: 4789
+    - run:
+        collectorName: check-data-dir-symlink
+        command: sh
+        args:
+          - -c
+          - |
+            [ -d "{{ .DataDir }}" ] && [ -L "{{ .DataDir }}" ] && echo "{{ .DataDir }} is a symlink" || echo "{{ .DataDir }} is not a symlink"
```

Comment:
So if datadir isn't a directory, we print "not a symlink." That's probably fine if Ethan's changes validate that datadir is a directory. Does it?

Reply:
That is a good point. It may be that someone just created a file there with the `embedded-cluster` name.

Reply:
If the data dir is a file then the installation bails out as it can't turn the file into a directory in order to materialize:

```
root@red-water:/home/developer# echo xxx >/var/lib/embedded-cluster
root@red-water:/home/developer# ./embedded-cluster install
unable to create embedded-cluster tmp dir: mkdir /var/lib/embedded-cluster: not a directory
root@red-water:/home/developer#
```